src/controller/CHIPDeviceController.cpp

```diff
@@ -215,46 +210,67 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR DeviceController::GenerateOperationalCertificates(const ByteSpan & CSR, NodeId deviceId, MutableByteSpan & cert)
+CHIP_ERROR DeviceController::GenerateOperationalCertificates(const ByteSpan & noc, MutableByteSpan & cert)
 {
-    // This code requires about 2K RAM to generate the certificates.
+    // This code requires about 1K RAM to generate the certificates.
```

Comment:
The memory requirement is because the caller (CHIPDeviceController) has to allocate this buffer so that the OperationalCredentialsDelegate can fill in the generated buffer in it.

Reply:
gotcha , Thanks !